src/services/playerData.js

```diff
@@ -0,0 +1,11 @@
+import { client, checkError } from './client';
+
+export async function getPlayers() {
+  let request = await client.from('players').select().order('name');
+  return checkError(request);
+}
+
+export async function getPlayerById(id) {
+  let request = await client.from('players').select('*, teams(*)').match({ id });
```

Comment:
`.single()`  after the `.match()`  would return a single object instead of an array containing a single object. Would allow you to have to use data[0] later on.
https://supabase.com/docs/reference/javascript/single